src/treq/client.py

```diff
@@ -194,6 +198,13 @@ def request(self, method, url, **kwargs):
                     b'content-type', [b'application/x-www-form-urlencoded'])
                 data = urlencode(data, doseq=True)
             bodyProducer = self._data_to_body_producer(data)
+        elif has_json:
+            # If data is sent as json, set Content-Type as 'application/json'
+            headers.setRawHeaders(
+                b'content-type', [b'application/json; charset=UTF-8'])
+            content = kwargs.get('json')
```

Comment:
It's not necessary to use `.get()` here, since the key is known to be present. You can use subscript syntax.